### PR TITLE
Update fitz.i related to #2365

### PR DESCRIPTION
**Pull request by JorjMcKie**
This pertains to Page method `get_drawings()`. 
The set of keys for "fs" paths already is complete by definition - so only fill and stroke paths require special attention. Each type should have the complete list of keys as a "fs" path would have, so we generate `None` values where necessary for "f" and "s" paths only. In view of performance, this is probably faster then subclassing `dict` and giving it the `__missing__()` method to achieve the same thing.

Unrelated to this fix:
When dealing with Optional Content Properties, the array of `/Locked` optional content groups is now also supported for input and output.

### CODE REVIEW

**Review by julian-smith-artifex-com — APPROVED** [review]

File: fitz/helper-fields.i [hunk 1]
```diff
@@ -471,10 +471,10 @@ void JM_get_widget_properties(fz_context *ctx, pdf_annot *annot, PyObject *Widge
         SETATTR_DROP(Widget, "field_type_string",
                 JM_UnicodeFromStr(JM_field_type_text(field_type)));
 
-        #if FZ_VERSION_MAJOR == 1 && FZ_VERSION_MINOR <= 22
+        #if FZ_VERSION_MAJOR == 1 && FZ_VERSION_MINOR < 22
```
Comment (julian-smith-artifex-com) [inline fitz/helper-fields.i:474]: Are you sure this change is correct? We run daily tests with MuPDF 1.21, 1.22 and master (which has version 1.23), and they've been building and passing ok for a while. Maybe your mupdf master needs updating to get the change of its version to 1.23?

File: fitz/helper-fields.i [hunk 2]
```diff
@@ -668,10 +677,10 @@ void JM_set_widget_properties(fz_context *ctx, pdf_annot *annot, PyObject *Widge
     value = GETATTR("field_name");
     if (value != Py_None) {
         char *name = JM_StrAsChar(value);
-        #if FZ_VERSION_MAJOR == 1 && FZ_VERSION_MINOR <= 22
+        #if FZ_VERSION_MAJOR == 1 && FZ_VERSION_MINOR < 22
             char *old_name = pdf_field_name(ctx, annot_obj);
         #else
-            char *old_name = pdf_load_field_name(ctx, annot_obj);
+        char *old_name = pdf_load_field_name(ctx, annot_obj);
```
Comment (julian-smith-artifex-com) [inline fitz/helper-fields.i:683]: Similar comment to above.

**Review by julian-smith-artifex-com — COMMENTED** [review]
.

File: fitz/helper-fields.i (hunk 1 above)
Comment (JorjMcKie) [reply fitz/helper-fields.i:474]: I only did this because of compile errors.

Comment (julian-smith-artifex-com) [reply fitz/helper-fields.i:474]: Great.

**Pull request merged**